lib/plug/csrf_protection.ex

```diff
@@ -103,17 +103,34 @@ defmodule Plug.CSRFProtection do
 
   def init(opts), do: opts
 
-  def call(conn, _opts) do
+  def call(conn, opts) do
+    opts = verify_opts opts
```

Comment:
Everyone should call init, if they don't, it is a bug. However, I don't think verifying the options before hand is the best way to go. I would change the conditional below:

    if opts[:with] == :exception do

to:

    case Keyword.get(opts, :with, :exception) do
      :exception -> ...
      :clear_session -> ...
      other -> raise "unknown :with option in forgery protection: #{inspect other}"
    end